pym/bob/cmds/build/share.py

```diff
@@ -0,0 +1,296 @@
+# Bob build tool
+# Copyright (C) 2021  Jan Klötzke
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from ...errors import BuildError
+from ...tty import WarnOnce
+from ...utils import asHexStr, hashDirectoryWithSize, isWindows
+import errno
+import os, os.path
+import json
+import shutil
+import sys
+import tempfile
+
+warnRepoSize = WarnOnce("The shared repository is over its quota. Run 'bob clean --shared' to free disk space!")
+warnGcDidNotHelp = WarnOnce("The automatic garbage collection of the shared repository was unable to free enough space. Run 'bob clean --shared' manually.")
+
+if sys.platform == 'win32':
```

Comment:
Because `isWindows()` includes the MSYS platform where the `msvcrt` module is not available. MSYS does implement the `fcntl.flock()` function so it should work... :crossed_fingers: